flytestdlib/storage/stow_store.go

```diff
@@ -291,7 +291,17 @@ func (s *StowStore) List(ctx context.Context, reference DataReference, maxItems
 	if err == nil {
 		results := make([]DataReference, len(items))
 		for index, item := range items {
-			results[index] = DataReference(item.URL().String())
+			// We don't use `item.URL()` because e.g. listing a google cloud storage
+			// bucket `gs://some-bucket/...` will result in items with URLs like
+			// `google://storage.googleapis.com/download/storage/v1/b/some-bucket/...`
+			// which subsequently cannot be found by the stow store.
+
+			u := url.URL{
+				Scheme: protocol,
+				Host:   containerName,
+				Path:   item.Name(),
+			}
+			results[index] = DataReference(u.String())
```

Comment:
Can we fix this in our fork of stow? It looks like we change the scheme [here](https://github.com/flyteorg/stow/blame/632adef717d32cc01727f9ec5b064c0505bbffb9/google/item.go#L84), which is used in gcs implementation of [convertToStowItem](https://github.com/flyteorg/stow/blob/master/google/container.go#L152-L176).

Reply:
Will do, overlooked we use a fork of stow 👌